Mono.Cecil/AssemblyReader.cs

```diff
@@ -875,6 +898,9 @@ TypeDefinition ReadType (uint rid)
 			if (IsNested (attributes))
 				type.DeclaringType = GetNestedTypeDeclaringType (type);
 
+			if (module.MetadataKind != MetadataKind.Ecma335)
```

Comment:
Can you add an extension method `IsWindowMetadata` on ModuleDefinition in ModuleDefinition.cs:Mixin that does this check, and use it everywhere instead of this check?